tools/scripts/script.ps1

```diff
@@ -0,0 +1,230 @@
+# ----------------------------------------------------------------------------------
+#
+# Copyright Microsoft Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------------
+
+param(
+    [Parameter(Mandatory = $false, Position = 0, HelpMessage = "To uninstall. By default false.")]
+    [switch]$uninstall
+)
+
+<#################################################
+#
+#               Helper functions
+#
+#################################################>
+
+function Get-PowerShellVersion{
+    if ($PSVersionTable.PSVersion.Major -lt 7){
+        Write-Warning "[For the best experience please install PowerShell version 7.]`n"
+    }
+}
+
+function Save-AIShellFromUrl{
+    # using placeholder urls until the GitHub release files are ready
+    $osxArm64Url = "osxArm64Url"
+    $osxX64Url = "osxX64Url"
+    $winArm64Url = "winArm64Url"
+    $winX64Url = "winX64Url"
+    $moduleUrl = "moduleUrl"
+
+    if (!$moduleUrl) {
+        return
+    }
+    
+    $destinationDirectory = (Join-Path (Join-Path (Join-Path $env:userprofile ".azure") "bin") "aishell")
```

Comment:
May be better to save in the `.aish` folder instead of `.azure`

Reply:
We should save to the proposed locations for Mac and Windows
- `$env:LOCALAPPDATA\Programs\AIShell` for Windows
- `/usr/local/AIShell` for Mac
Since there are too many distributions of Linux to check for we can encourage Linux users to go directly to the GitHub repo and download the zip package directly. So if a Linux user is trying to use the install script, please error and say "This install script is only compatible with Windows and Mac systems, if you are wanting to install on Linux please download the package directly from the GitHub repo at aka.ms/AIShell-Repo"